Add Military Features 10.x Tools to Repo

csmoore opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

Moving GP tools included with the ArcGIS 10.X-only Military Features template into this repo (and then will deprecate that repo): https://github.com/Esri/military-feature-toolbox

<!--- Describe your changes in detail -->

Adding the toolbox, scripts, and required executable to `military_features` folder and added unit tests to `utils\test\military_features_tests`

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Part of maintenance tasks for the template

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Manual and unit tests

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.

mfunk commented:
@csmoore Taking a look now...

mfunk commented:
**Delete All Features From Workspace** needs an output parameter defined (assuming it will be the same input workspace). All other tools are OK here.

![image](https://user-images.githubusercontent.com/3093182/30486279-e0f00f4e-99e4-11e7-8b62-ece1938a04fb.png)

mfunk commented on military_features/scripts/AppendMilitaryFeatures.py:
```diff
@@ -0,0 +1,123 @@
+    # Get the tool error messages 
+    msgs = arcpy.GetMessages() 
+    arcpy.AddError(msgs) 
+    print msgs
```
2.x to 3.x: print as function instead of statement: print(msgs)

csmoore replied:
These are 10.X/Python 2 tools with no possibility of Pro use

mfunk commented on military_features/scripts/CalcSIDCField.py:
```diff
@@ -0,0 +1,226 @@
+    # Get affiliation, needed because SIDC affiliation cannot always be derived from feature attributes
+    affiliation = arcpy.GetParameterAsText(4)
+    if (not (affiliation == "")) and (not affiliation is None) and (not affiliation in DictionaryConstants.validAffiliations) :
+        if (affiliation <> "NOT_SET") :
```
Syntax error here with "<>": suggest "if (affiliation is not "NOT_SET") :" a little more pythonic too  :)

csmoore replied:
At Python 3 (see note above about Python 2 only tools) it is a syntax error (no idea why I didn't use != ) not sure "is not" will work with constants. 

One note: for the toolbox/scripts I avoided making any changes to these to avoid having to rigorously retest - but I can make these changes if needed

mfunk commented:
Not sure if this is a problem or not, but have a missing default file in **Append Message File from Military Feature Workspace**.

![image](https://user-images.githubusercontent.com/3093182/30488194-8b11825a-99ea-11e7-988c-e606cf940022.png)

mfunk commented:
Also getting error running the Military Features tests on Desktop 10.3.1:

```
Test success: False

RESULTS =================================================
Number of tests run: 6
Number succeeded: 3
Number of errors: 3
Number of failures: 0
Number of tests skipped: 0
Percent passing: 0.0
=========================================================

ERRORS ==================================================
test_append_military_features (military_features_tests.AppendMilitaryFeaturesTestCase.AppendMilitaryFeaturesTestCase)
Traceback (most recent call last):
  File "D:\GitHub\solutions-geoprocessing-toolbox\utils\test\military_features_tests\AppendMilitaryFeaturesTestCase.py", line 95, in test_append_military_features
    toolOutput = arcpy.AppendMilitaryFeatures_MFT(self.inputPointsFC, \
AttributeError: 'module' object has no attribute 'AppendMilitaryFeatures_MFT'

test_calculate_rep_rule (military_features_tests.CalculateRepRuleTestCase.CalculateRepRuleTestCase)
Traceback (most recent call last):
  File "D:\GitHub\solutions-geoprocessing-toolbox\utils\test\military_features_tests\CalculateRepRuleTestCase.py", line 106, in test_calculate_rep_rule
    toolOutput = arcpy.CalculateRepresentationRuleField_MFT(inputRepRuleFC, \
AttributeError: 'module' object has no attribute 'CalculateRepresentationRuleField_MFT'

test_calculate_sidc (military_features_tests.CalculateSidcTestCase.CalculateSidcTestCase)
Traceback (most recent call last):
  File "D:\GitHub\solutions-geoprocessing-toolbox\utils\test\military_features_tests\CalculateSidcTestCase.py", line 104, in test_calculate_sidc
    toolOutput = arcpy.CalcSymbolIDField_MFT(inputCalcSidcFC, sidcField, \
AttributeError: 'module' object has no attribute 'CalcSymbolIDField_MFT'
END OF TEST =========================================
'One or more tests failed'
```

I looked at the tools and can't find:
* **AppendMilitaryFeatures**
* **CalculateRepresentationRuleField**
* **CalcSymbolIDField**

The tools I have in 10.3.1 are:
![image](https://user-images.githubusercontent.com/3093182/30489310-c790fa52-99eb-11e7-84f0-cc2910cb7a5b.png)


Perhaps the toolbox was built/modified in a later release?

csmoore commented:
Fixed Toolbox version issue (set back to 10.3) and added missing output param in Delete All Features From Workspace:

![image](https://user-images.githubusercontent.com/3090809/30546992-28122a82-9c5c-11e7-9679-72826b07bc21.png)

mfunk commented:
So far so good. Got more tools and all have outputs:

![image](https://user-images.githubusercontent.com/3093182/30547992-c7c2af82-9c45-11e7-851a-a8b1dc81611c.png)

mfunk commented:
Mil Features tests running without errors.

```
RESULTS =================================================
Number of tests run: 24
Number succeeded: 23
Number of errors: 0
Number of failures: 1
Number of tests skipped: 0
Percent passing: 0.0
=========================================================

FAILURES ================================================
testRouteAssetsToBasesAGOL (distance_to_assets_tests.DistanceToAssetsRouteAssetsToBasesAGOLTestCase.DistanceToAssetsRouteAssetsToBasesAGOLTestCase)
```

csmoore commented:
@mfunk - thanks for reviewing - I had already started some of the code cleanup request - so just included that in new PR #601
